PWGCF/TwoParticleCorrelations/Tasks/r2p2-4-id.cxx

```diff
@@ -0,0 +1,437 @@
+  std::shared_ptr<TH1> h1d_1p[2] = {histos.template get<TH1>(HIST("h1d_n1_ptP")), histos.template get<TH1>(HIST("h1d_n1_ptM"))};
+  for (auto track1 : tracks) {
+    //-----Single Particle Distribution----------------------------------------
+    //-----PION PID------------------------------------------------------------
+    histos.fill(HIST("h1d_n1_pt"), track1.pt());
+    histos.fill(HIST("h2d_n1_tof1"), track1.p(), track1.beta());
+    histos.fill(HIST("h2d_n1_tpc1"), track1.p(), track1.tpcSignal());
+    if (!pidarray<decltype(track1)>[p](track1))
+      continue;
+    histos.fill(HIST("h2d_n1_tof2"), track1.p(), track1.beta());
+    histos.fill(HIST("h2d_n1_tpc2"), track1.p(), track1.tpcSignal());
+    //-------------------------------------------------------------------------
+    histos.fill(HIST("h1d_n1_phi"), track1.phi());
+    histos.fill(HIST("h1d_n1_eta"), track1.eta());
+    sign = (track1.sign() + 1) / 2;
+    h1d_1p[sign]->Fill(track1.pt(), 1.0 / (2.0 * constants::math::PI * track1.pt()));
+    h2d_1p[sign][0]->Fill(track1.eta(), track1.phi());
+    h2d_1p[sign][1]->Fill(track1.eta(), track1.phi(), track1.pt());
+    etabin1 = (track1.eta() + 0.8) * 15;
+    phibin1 = 36 * track1.phi() / (2 * constants::math::PI);
+    if ((etabin1 < 0) || (etabin1 >= 24) || (phibin1 < 0) || (phibin1 >= 36))
+      continue;
+    mult++;
+    //------------------------------------------------------------------------
+    for (auto track2 : tracks) {
+      //-----Two Particle Distribution------------------------------------------
+      if (track1.index() == track2.index())
+        continue; // No auto-correlations
+      //-----PION PID----------------------------------------------------------
+      if (!pidarray<decltype(track2)>[p](track2))
+        continue;
+      //-----------------------------------------------------------------------
+      etabin2 = (track2.eta() + 0.8) * 15;
+      phibin2 = 36 * track2.phi() / (2 * constants::math::PI);
+      if ((etabin2 < 0) || (etabin2 >= 24) || (phibin2 < 0) || (phibin2 >= 36))
+        continue;
+      sign = (track1.sign() + track2.sign() + 2) / 2;
+      h2d_2p[sign][0]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5);
+      h2d_2p[sign][1]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5, track2.pt());
+      h2d_2p[sign][2]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5, track1.pt());
+      h2d_2p[sign][3]->Fill(36 * etabin1 + phibin1 + 0.5, 36 * etabin2 + phibin2 + 0.5, track1.pt() * track2.pt());
```

Comment:
I would suggest to have the array at the histogram level and not at the full histograms collection
With the current approach it is not easy to know which histogram is being filled which in case of issues is really difficult to debug

I meant something like
```
std::vector<std::vector<TH2F*>> h2d_n2_eta1Phi1Eta2Phi2;
```
which will be addressed as
```
h2d_n2_eta1Phi1Eta2Phi2[tag_track1][tag_track2]->Fill(...);
```

Please, take this as a suggestion which will make your code more readable and easy to debug in case of need
But it is not mandatory